doc/developer-documentation.md

```diff
@@ -69,13 +71,18 @@ jobs:
       tree:
         - <tree-name>
         - !<tree-name2>
+        - <tree-name3>:<branch-name>
 ```
 Here is the description of each field:
 - **`template`**: A `jinja2` template should be added to the [`config/runtime`](https://github.com/kernelci/kernelci-pipeline/tree/main/config/runtime) directory. This template will be used to generate the test definition.
 - **`kind`**: The `kind` field specifies the type of job. It should be `kbuild` for build jobs, `job` for a test suite, and `test` for a single test case.
 - **`image`**: The `image` field specifies the Docker image used for building and running the test. This field is optional. For example, LAVA test jobs use an image defined in the test definition template instead.
 - **`params`**: The `params` field includes parameters for building the kernel (for `kbuild` jobs) or running the test. These parameters can include architecture, compiler, defconfig options, job timeout, etc.
-- **`rules`**: The `rules` field defines job rules. If a test should be scheduled for a specific kernel tree, branch, or version, these rules can be specified here. The rules prefixed with `!` exclude the specified condition from job scheduling. For example, in the given scenario, the scheduler does not schedule a job if an event is received for the kernel tree `tree-name2`.
+- **`rules`**: The `rules` field defines job rules. If a test should be scheduled for a specific kernel tree, branch, or version, these rules can be specified here. The rules prefixed with `!` exclude the specified condition from job scheduling. For example, in the given scenario, the scheduler does not schedule a job if an event is received for the kernel tree `tree-name2`. It is also possible to schedule job for a specific branch of a tree. For example, a rule `tree:branch` can be mentioned under
+`tree` or `branch` section to define that, still accepting the `!` prefix to ensure jobs can be scheduled on all branches from a given tree, except a few select ones.
+Rules can be defined in relation to any attribute present in the parent node (such as `defconfig` or `fragments` or even `arch` for build jobs or
+test jobs such as `kunit`).
```

Comment:
```suggestion
Rules can be defined in relation to any attribute present in the parent node (such as `defconfig` or `fragments`, or even `arch` for test jobs such as `kunit`).
```

`arch` can't be used for build jobs, as they're created from a checkout node, which doesn't have this attribute.

Reply:
I see `params.arch` in a couple of build jobs such as https://github.com/kernelci/kernelci-pipeline/blob/main/config/pipeline.yaml#L51

Reply:
Indeed, but rules don't take params into account, only node data; moreover, having an `arch` rule for build jobs wouldn't make sense as this is indeed a job parameter, so we're sure it won't run for any other arch than the one defined in `params.arch`.

Reply:
Okay. Isn't the same applicable for `fragments` and `defconfig`?
Those are also `params` attributes.

Reply:
Yes, you're absolutely right! kbuild nodes can only use tree/branch/version rules (simply put, what is available from checkout nodes) indeed.

Reply:
Okay, thanks for clarifying.
Fixed the line.